src/jsonata.js

```diff
@@ -964,7 +965,7 @@ var jsonata = (function() {
                 env = createFrameFromTuple(environment, tuple);
             }
             var value = yield * evaluate(expr.lhs[entry.exprIndex][1], context, env);
-            if(typeof value !== 'undefined') {
+            if(typeof value !== 'undefined' && key !== 'undefined') {
```

Comment:
Ah, I've just noticed that you're comparing with the string 'undefined' rather than the undefined type.  This will break any JSON object that has a property name 'undefined' (it could happen!).
E.g. the following will break under this PR: https://try.jsonata.org/3KdIrw3DC

Reply:
Thanks @andrew-coleman that is a good point.  I refactored the code to just check the key for the undefined type and removed the string check.